src/transformer.ts

```diff
@@ -1,193 +1,581 @@
-                ?.replace(/\//g, '/xmlns:')}`;
-            const dataNode = doc.get(path, NAMESPACES);
-
-            if (dataNode) {
-                const text = dataNode.text();
-
-                // Very crude URL checker which is fine for now,
-                // because at this point we don't expect anything other than jr://
-                if (/^[a-zA-Z]+:\/\//.test(text)) {
-                    const value = getMediaPath(mediaMap, text);
-                    const escapedText = escapeURLPath(text);
-
-                    dataNode.attr({ src: value }).text(escapedText);
+    const binaryBindings = Array.from(
+        doc.querySelectorAll(':root > head > model > bind[type="binary"]')
+    );
+
+    binaryBindings.forEach((bind) => {
+        const nodeset = bind.getAttribute('nodeset');
+
+        if (nodeset) {
+            const expression = `/h:html/h:head/xmlns:model/xmlns:instance/xmlns:${nodeset
+                .trim()
+                .replace(/^\//, '')
+                .replace(/\//g, `/xmlns:`)}`;
+            const dataNodes = getNodesByXPathExpression(doc, expression);
+
+            dataNodes.forEach((dataNode) => {
+                if (dataNode instanceof Element) {
+                    const text = dataNode.textContent?.trim() ?? '';
+
+                    // TODO (2022-12-30): The comment below is concerning! Why would
+                    // we not make the pattern more restrictive if we know what we
+                    // do expect to match?
+
+                    // Very crude URL checker which is fine for now,
+                    // because at this point we don't expect anything other than jr://
+                    if (/^[a-zA-Z]+:\/\//.test(text)) {
+                        const src = getMediaPath(mediaMap, text);
+
+                        dataNode.setAttribute('src', src);
+                        dataNode.textContent = escapeURLPath(text);
+                    } else if (text !== '') {
+                        dataNode.setAttribute('src', escapeURLPath(text));
+                        dataNode.textContent = escapeURLPath(text);
+                    }
                 }
+            });
+        }
+    });
+};
+
+/**
+ *
+ * @param doc
+ */
+const transformAppearances = (doc: Document) => {
+    const appearanceElements =
+        doc.querySelectorAll<HTMLElement>('[data-appearances]');
+
+    appearanceElements.forEach((element) => {
+        const appearances =
+            element.dataset.appearances?.trim().toLowerCase().split(/\s+/) ??
+            [];
+        const classes = appearances.flatMap((appearance) => {
+            const results = [`or-appearance-${appearance}`];
+
+            // Convert deprecated appearances, but leave the deprecated ones.
+            if (appearance === 'horizontal') {
+                results.push('or-appearance-columns');
+            }
+
+            if (appearance === 'horizontal-compact') {
+                results.push('or-appearance-columns-pack');
```

Comment:
In some haste, my original reimplementation mistakenly had this line as `results.push('appearance-columns-pack')`. I didn't catch the mistake until a few days ago. I had originally left out the WHO VA and SOAR forms because even with these performance improvements I felt they'd slow down my progress. After rebasing the snapshots commit and capturing the first baseline benchmarks, I discovered my mistake when the WHO VA form's snapshot tests failed.

My point in mentioning this is that even with all of my efforts to thoroughly match the semantics and logical flow, it was the safeguard of fixtures and snapshot tests that caught the potential regression! (I could make similar comments on several other parts of this PR, but this one stood out to me because it was one of the last I caught and I had already convinced myself there would be no more.)